sphinx/ext/doctest.py

```diff
@@ -103,10 +106,40 @@ def run(self):
             for option in option_strings:
                 if (option[0] not in '+-' or option[1:] not in
                         doctest.OPTIONFLAGS_BY_NAME):  # type: ignore
-                    # XXX warn?
+                    self.state.document.reporter.warning(
+                            "missing '+' or '-' in '%s' option." % option, 
+                            line=self.lineno)
                     continue
                 flag = doctest.OPTIONFLAGS_BY_NAME[option[1:]]  # type: ignore
                 node['options'][flag] = (option[0] == '+')
+        if self.name == 'doctest' and 'pyversion' in self.options:
+            try:
```

Comment:
Could you separate this into a single function and add some test?
It seems the evaluation code are not tested well.